duffy/legacy/main.py

```diff
@@ -29,106 +31,160 @@
+        response = httpx.put(
+            "%s/api/v1/sessions/%s" % (obtain_destination(), ssid),
+            json={"active": False},
+            auth=(cred.username, cred.password),
+        )
+        if response.status_code == 200:
+            return JSONResponse(status_code=200, content="Done")
+        elif (
+            response.status_code == HTTP_401_UNAUTHORIZED
+            or response.status_code == HTTP_403_FORBIDDEN
+        ):
+            return JSONResponse(status_code=403, content={"msg": "Invalid duffy key"})
+        else:
+            return JSONResponse(status_code=200, content="Failed to return nodes on failure")
     else:
-        return JSONResponse(status_code=200, content="Failed to return nodes on failure")
+        return JSONResponse(status_code=200, content="Some parameters are absent")
 
 
-@app.post("/Inventory")
-async def get_all_nodes(data: InventoryRequestModel):
-    username, password, dest = obtain_credentials(data.key)
-    response = httpx.get("%s/api/v1/nodes" % dest, auth=(username, password))
-    if response.status_code == 200:
-        nodelist = response.json()["nodes"]
-        retnlist = []
-        for indx in nodelist:
-            retnlist.append(
-                [
-                    indx["id"],  # sch.data['id']
-                    indx["hostname"],  # sch.data['hostname']
-                    indx["ipaddr"],  # sch.data['ip']
-                    indx["type"],  # sch.data['chassis']
-                    0,  # sch.data['used_count']
-                    indx["state"],  # sch.data['state']
-                    indx["comment"],  # sch.data['comment']
-                    indx["distro_type"],  # sch.data['distro']
-                    indx["distro_version"],  # sch.data['rel']
-                    indx["distro_version"],  # sch.data['ver']
-                    None,  # sch.data['arch']
-                    None,  # sch.data['pool']
-                    None,  # sch.data['console_port']
-                    None,  # sch.data['flavor']
-                ]
-            )
-        return JSONResponse(status_code=200, content=retnlist)
-    elif (
-        response.status_code == HTTP_401_UNAUTHORIZED or response.status_code == HTTP_403_FORBIDDEN
-    ):
-        return JSONResponse(status_code=403, content={"msg": "Invalid duffy key"})
+@app.get("/Inventory")
+async def get_nodes(
+    key: Optional[str] = None, cred: Optional[Credentials] = Depends(obtain_credentials)
+):
+    if key:
+        response = httpx.get(
+            "%s/api/v1/sessions" % obtain_destination(), auth=(cred.username, cred.password)
+        )
+        if response.status_code == 200:
+            retnlist = []
+            sesslist = response.json()["sessions"]
+            for indx in sesslist:
+                nodelist = indx["nodes"]
+                for indx in nodelist:
+                    retnlist.append(
+                        [
+                            indx.get("id"),  # sch.data['id']
```

Comment:
What's `sch`?

Reply:
In an attempt to keep myself from getting fired for coming up with this strange list, I wanted to make it clear that it wasn't my idea to pass things as a list when a tuple or a dictionary could have better served the purpose - I made links to the place I copied this from i.e. the https://github.com/CentOS/duffy/blob/stale/master/duffy/api_v1/views.py#L174-L187 parts of the `views.py` for the legacy Duffy. I suggest for it to stay in order to be a good reference of how the things are arranged in the list.

Reply:
👍